### PR TITLE
Add Axolstudio Foundation Gamma

### PR DESCRIPTION
**Pull request by kb-elmo**
## Description

Added support for the Axolstudio Foundation TKL PCB

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/axolstudio/foundation_gamma/info.json [hunk 1]
```diff
@@ -0,0 +1,193 @@
+{
+    "keyboard_name": "Foundation Gamma", 
+    "url": "https://axolstudio.ca/foundation-gamma", 
+    "maintainer": "kb-elmo", 
+    "width": 18.25, 
+    "height": 6.5,
```
Comment (drashna) [inline keyboards/axolstudio/foundation_gamma/info.json:6]:
```suggestion
```

File: keyboards/axolstudio/foundation_gamma/rules.mk [hunk 2]
```diff
@@ -0,0 +1,22 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
+BOOTLOADER = atmel-dfu
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
```
Comment (drashna) [inline keyboards/axolstudio/foundation_gamma/rules.mk:10]: There is no "full" bootmagic anymore, only the "lite" option, which can be turned on with just "yes".
```suggestion
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
```

File: keyboards/axolstudio/foundation_gamma/rules.mk [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
```
Comment (drashna) [inline keyboards/axolstudio/foundation_gamma/rules.mk:21]:
```suggestion
NKRO_ENABLE = no            # Enable N-key Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by kb-elmo** [comment]
must have missed the change requests.
updated the pr now :)

**Comment by kb-elmo** [comment]
bump :)

**Review by drashna — APPROVED** [review]

File: keyboards/axolstudio/foundation_gamma/config.h [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+/* CAPS LED */
+#define LED_CAPS_LOCK_PIN B7
+
+/* Debounce reduces chatter (unintended double-presses) - set 0 if debouncing is not needed */
+#define DEBOUNCE 5
+
+/* disable these deprecated features by default */
+#define NO_ACTION_MACRO
+#define NO_ACTION_FUNCTION
```
Comment (fauxpark) [inline keyboards/axolstudio/foundation_gamma/config.h:48]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Pull request merged**